src/categorical_algebra/Sets.jl

```diff
@@ -97,21 +111,88 @@ codom(f::IdentityFunction) = f.dom
 
 (f::IdentityFunction)(x) = x
 
+Base.inv(f::IdentityFunction) = f
+
 function Base.show(io::IO, f::IdentityFunction)
   print(io, "id(")
   show_domains(io, f, codomain=false)
   print(io, ")")
 end
 
+""" An abstract type for bijections in **Set** that are implemented as wrappers
+for `SetFunction` objects.
+
+Each concrete subtype must store the wrapped `SetFunction` in the `func` field.
+"""
+abstract type AbsBijectionWrap{Dom<:SetOb,Codom<:SetOb,F<:SetFunction{Dom,Codom}} <:
+  Bijection{Dom,Codom} end
+
+unwrap(f::AbsBijectionWrap) = unwrap(f.func)
+unwrap(f::SetFunction) = f
+
+dom(f::AbsBijectionWrap) = dom(unwrap(f))
+codom(f::AbsBijectionWrap) = codom(unwrap(f))
+
+(f::AbsBijectionWrap)(x) = (unwrap(f))(x)
+
+# Developer's note: should I add in a `known_correct` parameter, asking
+# the caller to decide whether the SetFunction should be checked for bijectivity?
+@struct_hash_equal struct BijectionThinWrap{Dom<:SetOb, Codom<:SetOb,
+    F<:SetFunction{Dom,Codom}} <: AbsBijectionWrap{Dom, Codom, F}
+  func::F
+end
+
+BijectionThinWrap(f::BijectionThinWrap) = f
+
+function Base.show(io::IO, f::BijectionThinWrap)
+  replacerules = (r"^SetFunction" => "Bijection", r"^FinFunction" => "FinBijection")
+  print(io, replace(repr(f.func), replacerules...))
```

Comment:
Make sure to include a test for this so that we can detect breakage. I can imagine changing the "SetFunction" and this pattern not matching any more.